src/test/groovy/ai/toloka/client/v1/impl/AbstractClientImplSpec.groovy

```diff
@@ -0,0 +1,125 @@
+package ai.toloka.client.v1.impl
+
+import ai.toloka.client.v1.impl.transport.DefaultHttpClientConfiguration
+import ai.toloka.client.v1.operation.Operation
+import ai.toloka.client.v1.TolokaRequestIOException
+import ai.toloka.client.v1.TlkException
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+import groovy.json.JsonBuilder
+import org.apache.http.impl.client.HttpClientBuilder
+import org.apache.http.message.BasicHeader
+import org.mockserver.client.server.MockServerClient
+
+import static java.util.Collections.singletonList
+import static org.mockserver.model.HttpRequest.request
+import static org.mockserver.model.HttpResponse.response
+
+class TestEntity {
+    int id;
+
+    @JsonCreator
+    TestEntity(@JsonProperty("id") id) {
+        this.id = id
+    }
+}
+
+
+class AbstractClientImplSpec extends AbstractClientSpec {
+    def httpClient = HttpClientBuilder.create()
+            .setMaxConnTotal(2)
+            .setMaxConnPerRoute(2)
+            .setDefaultRequestConfig(DefaultHttpClientConfiguration.getDefaultRequestConfig())
+            .setUserAgent(DefaultHttpClientConfiguration.getUserAgent())
+            .setDefaultHeaders(singletonList(DefaultHttpClientConfiguration.getDefaultAuthorizationHeader('abc')))
+            .build()
+
+    def testFactory = new TolokaClientFactoryImpl(new URI('http://localhost:8083/api/'), httpClient)
+            .tune({ http ->  http.addHeader(new BasicHeader(headerName, headerValue)) })
+
+    def abstractClient = new AbstractClientImpl(testFactory) {}
+
+    def "return connection in pool; get with 404" () {
+        setup:
+        new MockServerClient('localhost', 8083)
+                .when(request('/api/v1/test-get/1'))
+                .respond(response(new JsonBuilder(test_map()).toString()).withStatusCode(404))
```

Comment:
lets put every method call to separate line, so it would be more readable 

```
        new MockServerClient('localhost', 8083)
                .when(request('/api/v1/test-get/1'))
                .respond(response(new JsonBuilder(test_map()).toString())
                .withStatusCode(404))
```

same for others

Reply:
Done